src/tests/compat_api.rs

```diff
@@ -551,4 +590,36 @@ mod when_no_message_received_within_inactivity_period {
         };
         assert!(peer1_lost || peer2_lost);
     }
+
+    #[test]
+    fn when_heartbeats_exchanging_tcp_peer_stays_alive() {
+        let (event_rx1, event_rx2, _s1, _s2) =
+            bootstrap_and_do_nothing(tcp_addr!("0.0.0.0:0"), true);
+        thread::sleep(Duration::from_millis(5 * INACTIVITY_TIMEOUT_MS));
```

Comment:
Can we not just wait until we receive lost peer from the channel rather than compulsorily waiting for this duration ? Seems a little wasteful otherwise to just block the test for this duration - think the timeout is pretty low for tests but we shouldn't rely on it unless we are testing the event is not fired before the timeout or something but probably that's not the case here.
We can ofc use [rx_timeout(inactivity_timeout + 10ms or something)](https://doc.rust-lang.org/std/sync/mpsc/struct.Receiver.html#method.recv_timeout) if we are concerned about hanging in case of a buggy production code.
Also check in other places if applicable.